integration_tests/testsuite/test_util.py

```
@@ -140,3 +141,18 @@ def _project_id():
 def generate_version():
     return 'integration-{0}'.format(
         datetime.datetime.now().strftime('%Y%m%d%H%m%S'))
+
+
+@retry(wait_fixed=10000, stop_max_attempt_number=4)
+def retrieve_url_for_version(version):
+    try:
+        # retrieve url of deployed app for test driver
+        url_command = ['gcloud', 'app', 'versions', 'describe',
+                       version, '--service',
+                       'default', '--format=json']
+        app_dict = json.loads(subprocess.check_output(url_command))
+        return app_dict.get('versionUrl')
+    except (subprocess.CalledProcessError, ValueError, KeyError):
+        logging.warn('Error encountered when retrieving app URL!')
+        return None
```

Should we throw here instead of returning None?